brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -83,7 +90,8 @@ public SpanId startNewSpan(String component, String operation) {
     }
 
     private SpanId getNewSpanId() {
-        Span parentSpan = spanAndEndpoint().state().getCurrentServerSpan().getSpan();
+        Span parentSpan = spanAndEndpoint().state().getCurrentLocalSpan();
```

Comment:
If we were to do this, we should probably keep parity with the logic in ClientTracer, copy/pasting here:

```java
        Span parentSpan = spanAndEndpoint().state().getCurrentLocalSpan();
        if (parentSpan == null) {
            ServerSpan serverSpan = spanAndEndpoint().state().getCurrentServerSpan();
            if (serverSpan != null) {
                parentSpan = serverSpan.getSpan();
            }
        }
```

for example, that addresses the possible null pointer